### PR TITLE
mess of code for `apply_vaccination.R` produces vaccinated matrix but…

### PR DESCRIPTION
**Pull request by geryan**
… needs to be tidied into function format.

@aarathybabu97 to tidy this into a working format - currently an irresponsible unworkable script...

### CODE REVIEW

**Comment by geryan** [comment]
(don't merge until tidied)

**Comment by aarathybabu97** [comment]
```r

library(conmat)
#> 
#> Attaching package: 'conmat'
#> The following objects are masked _by_ '.GlobalEnv':
#> 
#>     estimate_setting_contacts, extrapolate_polymod
 vaccination_effect_example_data
#> # A tibble: 17 x 4
#>    age_band coverage acquisition transmission
#>    <chr>       <dbl>       <dbl>        <dbl>
#>  1 0-4         0           0            0    
#>  2 5-11        0.782       0.583        0.254
#>  3 12-15       0.997       0.631        0.295
#>  4 16-19       0.965       0.786        0.469
#>  5 20-24       0.861       0.774        0.453
#>  6 25-29       0.997       0.778        0.458
#>  7 30-34       0.998       0.803        0.493
#>  8 35-39       0.998       0.829        0.533
#>  9 40-44       0.999       0.841        0.551
#> 10 45-49       0.993       0.847        0.562
#> 11 50-54       0.999       0.857        0.579
#> 12 55-59       0.996       0.864        0.591
#> 13 60-64       0.998       0.858        0.581
#> 14 65-69       0.999       0.864        0.591
#> 15 70-74       0.999       0.867        0.597
#> 16 75-79       0.999       0.866        0.595
#> 17 80+         0.999       0.844        0.556
 ngm_NSW <- generate_ngm(state_name = "NSW", 
                         age_breaks = c(seq(0, 80, by = 5),Inf),
                         R_target = 1.5
                         ) 
 ngm_NSW$all
#>                [0,5)      [5,10)     [10,15)     [15,20)     [20,25)
#> [0,5)    0.105119565 0.051693794 0.018301115 0.012344944 0.020170646
#> [5,10)   0.068608738 0.279823500 0.074443568 0.022489704 0.018470817
#> [10,15)  0.037908084 0.092171499 0.435578743 0.089168659 0.029762839
#> [15,20)  0.036167362 0.038954418 0.114311912 0.463723946 0.099489031
#> [20,25)  0.078474660 0.047367627 0.056730985 0.164217927 0.355841672
#> [25,30)  0.174037458 0.092433247 0.056430963 0.075740930 0.216164488
#> [30,35)  0.237782934 0.181278569 0.094748947 0.062961456 0.106331493
#> [35,40)  0.172589301 0.215722690 0.160755032 0.083585470 0.074694250
#> [40,45)  0.089320176 0.139028650 0.173414083 0.124108253 0.077564100
#> [45,50)  0.058762317 0.071928631 0.112003714 0.136730382 0.107613395
#> [50,55)  0.057150198 0.047331989 0.056492106 0.088509790 0.117735406
#> [55,60)  0.062405481 0.044441675 0.034300132 0.041874127 0.075259885
#> [60,65)  0.055314317 0.045688228 0.028647336 0.022348932 0.033502438
#> [65,70)  0.036352001 0.036766933 0.026361758 0.016669137 0.016122111
#> [70,75)  0.020397410 0.022002373 0.019632750 0.014386688 0.011379212
#> [75,80)  0.010544998 0.011428000 0.011178321 0.010409274 0.009563331
#> [80,Inf) 0.006863074 0.008225662 0.008229906 0.008529428 0.010440610
#>              [25,30)     [30,35)    [35,40)    [40,45)    [45,50)    [50,55)
#> [0,5)    0.040827924 0.054568563 0.03958914 0.02285643 0.01883850 0.02133527
#> [5,10)   0.031791489 0.063424572 0.08214487 0.05807597 0.03678757 0.03286735
#> [10,15)  0.027034826 0.044723268 0.08276140 0.09864450 0.06873067 0.04619231
#> [15,20)  0.041762231 0.040115330 0.05966314 0.09600786 0.10411747 0.07335998
#> [20,25)  0.147143683 0.080589849 0.07775703 0.09851643 0.13660264 0.14248669
#> [25,30)  0.370724616 0.199692714 0.13551791 0.12274320 0.13736642 0.17851697
#> [30,35)  0.234284402 0.340293388 0.22669651 0.16364867 0.13734763 0.15247877
#> [35,40)  0.119080307 0.200908178 0.28200189 0.21050120 0.14660475 0.12394650
#> [40,45)  0.081058229 0.103646931 0.16225570 0.22832891 0.17584568 0.12154226
#> [45,50)  0.082248591 0.077779789 0.09472019 0.14212208 0.20874924 0.16488064
#> [50,55)  0.104485823 0.079666014 0.07322202 0.08543103 0.13029698 0.21449673
#> [55,60)  0.106285265 0.096237279 0.07255062 0.06187200 0.07086463 0.12617193
#> [60,65)  0.062057704 0.088579751 0.08168306 0.05590877 0.04424561 0.05833971
#> [65,70)  0.024618671 0.045316781 0.06684441 0.05752257 0.03609761 0.03098885
#> [70,75)  0.011738155 0.017758023 0.03337584 0.04651249 0.03739811 0.02475502
#> [75,80)  0.008520691 0.008883585 0.01363228 0.02395430 0.03060585 0.02510187
#> [80,Inf) 0.012172600 0.012465506 0.01306234 0.01689492 0.02508046 0.03253169
#>             [55,60)    [60,65)    [65,70)    [70,75)    [75,80)   [80,Inf)
#> [0,5)    0.02411623 0.02183090 0.01573257 0.01351693 0.01585976 0.01757519
#> [5,10)   0.03217978 0.03277361 0.02703191 0.02022905 0.01993994 0.02756727
#> [10,15)  0.03490682 0.03221436 0.03267022 0.02758998 0.02327147 0.03067676
#> [15,20)  0.04504865 0.03178460 0.03140717 0.03253355 0.02899572 0.03137648
#> [20,25)  0.09880358 0.05777798 0.04270228 0.04469665 0.04519700 0.04599102
#> [25,30)  0.18024974 0.11869501 0.06845267 0.05741564 0.05828658 0.06408175
#> [30,35)  0.19078235 0.17899505 0.11399899 0.07543563 0.06423432 0.07197008
#> [35,40)  0.13790145 0.16182228 0.14581481 0.10374815 0.07261078 0.06639473
#> [40,45)  0.10285612 0.10774817 0.12063707 0.11848915 0.09145437 0.06484876
#> [45,50)  0.11155270 0.08782508 0.08625768 0.10264021 0.10783654 0.08474812
#> [50,55)  0.16798442 0.10539098 0.07650497 0.07851369 0.09656998 0.11420253
#> [55,60)  0.22891435 0.16929796 0.09773423 0.07239106 0.07467371 0.12180878
#> [60,65)  0.12215646 0.22316498 0.15549199 0.09060906 0.06502646 0.09452901
#> [65,70)  0.04821369 0.10556020 0.18827210 0.13340994 0.07343390 0.06098357
#> [70,75)  0.02440572 0.03996615 0.08584034 0.15111378 0.10203921 0.04899723
#> [75,80)  0.01859135 0.01968697 0.03156120 0.06768198 0.11351025 0.05799354
#> [80,Inf) 0.03226154 0.02766517 0.02625227 0.03608725 0.06836967 0.11055769
 ngmNSW_vaccination<- apply_vaccination(ngm = ngm_NSW,
                                          data=vaccination_effect_example_data,
                                          coverage_col=coverage,
                                          acquisition_col=acquisition,
                                          transmission_col=transmission)
 
 ngmNSW_vaccination$all
#>                [0,5)      [5,10)      [10,15)      [15,20)      [20,25)
#> [0,5)    0.105071450 0.041268418 0.0128581967 0.0067396830 0.0122468731
#> [5,10)   0.037491505 0.121901698 0.0285379582 0.0067096702 0.0061537505
#> [10,15)  0.014129531 0.027373154 0.1141941394 0.0181314192 0.0067875501
#> [15,20)  0.008736699 0.007504315 0.0194041756 0.0613715762 0.0146844355
#> [20,25)  0.026149702 0.012608027 0.0132961158 0.0299890509 0.0723011826
#> [25,30)  0.039169067 0.016664757 0.0089470298 0.0093703302 0.0297074724
#> [30,35)  0.047337345 0.028972562 0.0133275832 0.0068928464 0.0129615088
#> [35,40)  0.029586187 0.029741783 0.0195262877 0.0078672139 0.0078399272
#> [40,45)  0.014192723 0.017805248 0.0196106756 0.0108722203 0.0075614781
#> [45,50)  0.009220715 0.009078389 0.0125197289 0.0118664251 0.0103733582
#> [50,55)  0.008234127 0.005454506 0.0057618296 0.0070221752 0.0103697088
#> [55,60)  0.008832631 0.005024148 0.0034118606 0.0032441593 0.0064820816
#> [60,65)  0.008036111 0.005309878 0.0029278865 0.0017719525 0.0029643595
#> [65,70)  0.004989979 0.004047441 0.0025589497 0.0012531102 0.0013472239
#> [70,75)  0.002691888 0.002337143 0.0018404956 0.0010454330 0.0009151580
#> [75,80)  0.001396409 0.001219274 0.0010506820 0.0007584711 0.0007705162
#> [80,Inf) 0.001065868 0.001046808 0.0009224368 0.0007393222 0.0010046205
#>               [25,30)      [30,35)      [35,40)     [40,45)     [45,50)
#> [0,5)    0.0220890624 0.0276252318 0.0184870888 0.010280588 0.008314146
#> [5,10)   0.0093876135 0.0175336172 0.0209771351 0.014294663 0.008889493
#> [10,15)  0.0054472305 0.0084153215 0.0144016364 0.016544339 0.011323272
#> [15,20)  0.0054821537 0.0048801532 0.0067292264 0.010438635 0.011112733
#> [20,25)  0.0267702234 0.0135854193 0.0120765313 0.014831801 0.020123966
#> [25,30)  0.0455510264 0.0228832528 0.0141503255 0.012432422 0.013691311
#> [30,35)  0.0254001770 0.0346424824 0.0210507664 0.014540695 0.012085576
#> [35,40)  0.0110819524 0.0175438674 0.0227988639 0.016220235 0.011157849
#> [40,45)  0.0070184204 0.0083573009 0.0121581308 0.016451352 0.012463620
#> [45,50)  0.0070541969 0.0061834914 0.0069620178 0.010119250 0.014507173
#> [50,55)  0.0081773512 0.0057932208 0.0048963818 0.005531687 0.008300679
#> [55,60)  0.0081227086 0.0068505467 0.0047549644 0.003909904 0.004431609
#> [60,65)  0.0048714477 0.0064789638 0.0055031924 0.003625779 0.002834722
#> [65,70)  0.0018326059 0.0031537485 0.0042860182 0.003542124 0.002184567
#> [70,75)  0.0008438336 0.0011990446 0.0020800395 0.002784374 0.002195632
#> [75,80)  0.0006132346 0.0006040477 0.0008575903 0.001445665 0.001809177
#> [80,Inf) 0.0010400416 0.0010045964 0.0009804927 0.001217951 0.001752654
#>              [50,55)     [55,60)     [60,65)     [65,70)     [70,75)
#> [0,5)    0.008964482 0.009881913 0.009110679 0.006379003 0.005420899
#> [5,10)   0.007543057 0.007189686 0.007516619 0.006065500 0.004482484
#> [10,15)  0.007233748 0.005304056 0.005004636 0.004977189 0.004152338
#> [15,20)  0.007446718 0.004449380 0.003192575 0.003065736 0.003127818
#> [20,25)  0.019960763 0.013504169 0.008061515 0.005747543 0.005846802
#> [25,30)  0.016861041 0.016628686 0.011233684 0.006309535 0.005125332
#> [30,35)  0.012740691 0.015552465 0.014998988 0.009386369 0.006099047
#> [35,40)  0.009015662 0.009717507 0.011672226 0.010360777 0.007327071
#> [40,45)  0.008310474 0.006814312 0.007210541 0.007911682 0.007778735
#> [45,50)  0.011057245 0.007362132 0.005830884 0.005532454 0.006572056
#> [50,55)  0.012901223 0.010009299 0.006430865 0.004484154 0.004517467
#> [55,60)  0.007502278 0.013093284 0.010028036 0.005653867 0.004063284
#> [60,65)  0.003599222 0.007286070 0.013405457 0.009223532 0.005265019
#> [65,70)  0.001803817 0.002760274 0.006105364 0.010545812 0.007398920
#> [70,75)  0.001388856 0.001343735 0.002250599 0.004684271 0.008131637
#> [75,80)  0.001414644 0.001023179 0.001106463 0.001725713 0.003653141
#> [80,Inf) 0.002157663 0.002094645 0.001835337 0.001685971 0.002272630
#>              [75,80)    [80,Inf)
#> [0,5)    0.006436010 0.007751383
#> [5,10)   0.004455266 0.006822022
#> [10,15)  0.003515832 0.005149097
#> [15,20)  0.002800162 0.003351931
#> [20,25)  0.005905678 0.006654699
#> [25,30)  0.005104381 0.006180959
#> [30,35)  0.005076317 0.006114227
#> [35,40)  0.005092285 0.004907536
#> [40,45)  0.006065696 0.004559004
#> [45,50)  0.007074406 0.006038313
#> [50,55)  0.005697419 0.007527264
#> [55,60)  0.004218225 0.007836936
#> [60,65)  0.003735352 0.006169433
#> [65,70)  0.004028986 0.003712341
#> [70,75)  0.005499101 0.002865959
#> [75,80)  0.006202806 0.003442105
#> [80,Inf) 0.004355568 0.007819317
Created on 2022-04-01 by the reprex package (v2.0.1)
```

**Comment by aarathybabu97** [comment]
```r
library(conmat)
#> 
#> Attaching package: 'conmat'
#> The following objects are masked _by_ '.GlobalEnv':
#> 
#>     estimate_setting_contacts, extrapolate_polymod
 vaccination_effect_example_data
#> # A tibble: 17 x 4
#>    age_band coverage acquisition transmission
#>    <chr>       <dbl>       <dbl>        <dbl>
#>  1 0-4         0           0            0    
#>  2 5-11        0.782       0.583        0.254
#>  3 12-15       0.997       0.631        0.295
#>  4 16-19       0.965       0.786        0.469
#>  5 20-24       0.861       0.774        0.453
#>  6 25-29       0.997       0.778        0.458
#>  7 30-34       0.998       0.803        0.493
#>  8 35-39       0.998       0.829        0.533
#>  9 40-44       0.999       0.841        0.551
#> 10 45-49       0.993       0.847        0.562
#> 11 50-54       0.999       0.857        0.579
#> 12 55-59       0.996       0.864        0.591
#> 13 60-64       0.998       0.858        0.581
#> 14 65-69       0.999       0.864        0.591
#> 15 70-74       0.999       0.867        0.597
#> 16 75-79       0.999       0.866        0.595
#> 17 80+         0.999       0.844        0.556
 ngm_NSW <- generate_ngm(state_name = "NSW", 
                         age_breaks = c(seq(0, 80, by = 5),Inf),
                         R_target = 1.5
                         ) 
 
 ngm_NSW$all
#>                [0,5)      [5,10)     [10,15)     [15,20)     [20,25)
#> [0,5)    0.105119565 0.051693794 0.018301115 0.012344944 0.020170646
#> [5,10)   0.068608738 0.279823500 0.074443568 0.022489704 0.018470817
#> [10,15)  0.037908084 0.092171499 0.435578743 0.089168659 0.029762839
#> [15,20)  0.036167362 0.038954418 0.114311912 0.463723946 0.099489031
#> [20,25)  0.078474660 0.047367627 0.056730985 0.164217927 0.355841672
#> [25,30)  0.174037458 0.092433247 0.056430963 0.075740930 0.216164488
#> [30,35)  0.237782934 0.181278569 0.094748947 0.062961456 0.106331493
#> [35,40)  0.172589301 0.215722690 0.160755032 0.083585470 0.074694250
#> [40,45)  0.089320176 0.139028650 0.173414083 0.124108253 0.077564100
#> [45,50)  0.058762317 0.071928631 0.112003714 0.136730382 0.107613395
#> [50,55)  0.057150198 0.047331989 0.056492106 0.088509790 0.117735406
#> [55,60)  0.062405481 0.044441675 0.034300132 0.041874127 0.075259885
#> [60,65)  0.055314317 0.045688228 0.028647336 0.022348932 0.033502438
#> [65,70)  0.036352001 0.036766933 0.026361758 0.016669137 0.016122111
#> [70,75)  0.020397410 0.022002373 0.019632750 0.014386688 0.011379212
#> [75,80)  0.010544998 0.011428000 0.011178321 0.010409274 0.009563331
#> [80,Inf) 0.006863074 0.008225662 0.008229906 0.008529428 0.010440610
#>              [25,30)     [30,35)    [35,40)    [40,45)    [45,50)    [50,55)
#> [0,5)    0.040827924 0.054568563 0.03958914 0.02285643 0.01883850 0.02133527
#> [5,10)   0.031791489 0.063424572 0.08214487 0.05807597 0.03678757 0.03286735
#> [10,15)  0.027034826 0.044723268 0.08276140 0.09864450 0.06873067 0.04619231
#> [15,20)  0.041762231 0.040115330 0.05966314 0.09600786 0.10411747 0.07335998
#> [20,25)  0.147143683 0.080589849 0.07775703 0.09851643 0.13660264 0.14248669
#> [25,30)  0.370724616 0.199692714 0.13551791 0.12274320 0.13736642 0.17851697
#> [30,35)  0.234284402 0.340293388 0.22669651 0.16364867 0.13734763 0.15247877
#> [35,40)  0.119080307 0.200908178 0.28200189 0.21050120 0.14660475 0.12394650
#> [40,45)  0.081058229 0.103646931 0.16225570 0.22832891 0.17584568 0.12154226
#> [45,50)  0.082248591 0.077779789 0.09472019 0.14212208 0.20874924 0.16488064
#> [50,55)  0.104485823 0.079666014 0.07322202 0.08543103 0.13029698 0.21449673
#> [55,60)  0.106285265 0.096237279 0.07255062 0.06187200 0.07086463 0.12617193
#> [60,65)  0.062057704 0.088579751 0.08168306 0.05590877 0.04424561 0.05833971
#> [65,70)  0.024618671 0.045316781 0.06684441 0.05752257 0.03609761 0.03098885
#> [70,75)  0.011738155 0.017758023 0.03337584 0.04651249 0.03739811 0.02475502
#> [75,80)  0.008520691 0.008883585 0.01363228 0.02395430 0.03060585 0.02510187
#> [80,Inf) 0.012172600 0.012465506 0.01306234 0.01689492 0.02508046 0.03253169
#>             [55,60)    [60,65)    [65,70)    [70,75)    [75,80)   [80,Inf)
#> [0,5)    0.02411623 0.02183090 0.01573257 0.01351693 0.01585976 0.01757519
#> [5,10)   0.03217978 0.03277361 0.02703191 0.02022905 0.01993994 0.02756727
#> [10,15)  0.03490682 0.03221436 0.03267022 0.02758998 0.02327147 0.03067676
#> [15,20)  0.04504865 0.03178460 0.03140717 0.03253355 0.02899572 0.03137648
#> [20,25)  0.09880358 0.05777798 0.04270228 0.04469665 0.04519700 0.04599102
#> [25,30)  0.18024974 0.11869501 0.06845267 0.05741564 0.05828658 0.06408175
#> [30,35)  0.19078235 0.17899505 0.11399899 0.07543563 0.06423432 0.07197008
#> [35,40)  0.13790145 0.16182228 0.14581481 0.10374815 0.07261078 0.06639473
#> [40,45)  0.10285612 0.10774817 0.12063707 0.11848915 0.09145437 0.06484876
#> [45,50)  0.11155270 0.08782508 0.08625768 0.10264021 0.10783654 0.08474812
#> [50,55)  0.16798442 0.10539098 0.07650497 0.07851369 0.09656998 0.11420253
#> [55,60)  0.22891435 0.16929796 0.09773423 0.07239106 0.07467371 0.12180878
#> [60,65)  0.12215646 0.22316498 0.15549199 0.09060906 0.06502646 0.09452901
#> [65,70)  0.04821369 0.10556020 0.18827210 0.13340994 0.07343390 0.06098357
#> [70,75)  0.02440572 0.03996615 0.08584034 0.15111378 0.10203921 0.04899723
#> [75,80)  0.01859135 0.01968697 0.03156120 0.06768198 0.11351025 0.05799354
#> [80,Inf) 0.03226154 0.02766517 0.02625227 0.03608725 0.06836967 0.11055769
 ngmNSW_vaccination<- apply_vaccination(ngm = ngm_NSW,
                                          data=vaccination_effect_example_data,
                                          coverage_col=coverage,
                                          acquisition_col=acquisition,
                                          transmission_col=transmission)
 
 ngmNSW_vaccination$all
#>                [0,5)      [5,10)      [10,15)      [15,20)      [20,25)
#> [0,5)    0.105119565 0.041415002 0.0129140338 0.0067641599 0.0123159846
#> [5,10)   0.037307371 0.121904202 0.0285645081 0.0067007543 0.0061326848
#> [10,15)  0.014066363 0.027400969 0.1140515044 0.0181295475 0.0067433186
#> [15,20)  0.008746376 0.007547226 0.0195068515 0.0614462808 0.0146904759
#> [20,25)  0.026213030 0.012676188 0.0133718797 0.0300561320 0.0725761479
#> [25,30)  0.039139432 0.016654027 0.0089551653 0.0093331268 0.0296828409
#> [30,35)  0.047224276 0.028843660 0.0132783275 0.0068514758 0.0128942435
#> [35,40)  0.029658387 0.029699475 0.0194931754 0.0078702620 0.0078373666
#> [40,45)  0.014348456 0.017892836 0.0196573202 0.0109239826 0.0076079154
#> [45,50)  0.009326892 0.009146588 0.0125445423 0.0118912583 0.0104292650
#> [50,55)  0.008227322 0.005459022 0.0057386918 0.0069816301 0.0103489716
#> [55,60)  0.008720562 0.004975439 0.0033822201 0.0032062119 0.0064214720
#> [60,65)  0.007937307 0.005252417 0.0029007093 0.0017571835 0.0029353598
#> [65,70)  0.004990615 0.004043919 0.0025537857 0.0012539020 0.0013514406
#> [70,75)  0.002731726 0.002360755 0.0018553573 0.0010557173 0.0009305154
#> [75,80)  0.001418637 0.001231727 0.0010611722 0.0007673087 0.0007855674
#> [80,Inf) 0.001082024 0.001038983 0.0009155824 0.0007368224 0.0010050633
#>               [25,30)      [30,35)      [35,40)     [40,45)     [45,50)
#> [0,5)    0.0222091435 0.0277106450 0.0185211257 0.010281976 0.008325577
#> [5,10)   0.0094037394 0.0175136584 0.0208971437 0.014206263 0.008840651
#> [10,15)  0.0054569261 0.0084272839 0.0143670985 0.016466111 0.011271156
#> [15,20)  0.0054937581 0.0049263562 0.0067500769 0.010444467 0.011127634
#> [20,25)  0.0267364605 0.0136701171 0.0121511868 0.014803508 0.020165768
#> [25,30)  0.0453521335 0.0228054224 0.0142580381 0.012417593 0.013652754
#> [30,35)  0.0253106042 0.0343196027 0.0210630243 0.014620603 0.012055179
#> [35,40)  0.0111313571 0.0175321585 0.0226713275 0.016272580 0.011133958
#> [40,45)  0.0070831627 0.0084550495 0.0121940183 0.016500028 0.012484041
#> [45,50)  0.0071013516 0.0062691537 0.0070335065 0.010147704 0.014643026
#> [50,55)  0.0081822484 0.0058239546 0.0049314399 0.005532541 0.008289781
#> [55,60)  0.0080792173 0.0068291896 0.0047430110 0.003889413 0.004376421
#> [60,65)  0.0048440189 0.0064546720 0.0054835134 0.003608974 0.002805909
#> [65,70)  0.0018385051 0.0031592857 0.0042932097 0.003552484 0.002190141
#> [70,75)  0.0008551384 0.0012077048 0.0020911520 0.002802204 0.002213500
#> [75,80)  0.0006235542 0.0006069002 0.0008579951 0.001449695 0.001819689
#> [80,Inf) 0.0010439412 0.0009980039 0.0009634527 0.001198236 0.001747517
#>              [50,55)     [55,60)     [60,65)     [65,70)     [70,75)
#> [0,5)    0.009000097 0.009923317 0.009174214 0.006445361 0.005454771
#> [5,10)   0.007539263 0.007200220 0.007489224 0.006021985 0.004439042
#> [10,15)  0.007230506 0.005329761 0.005023389 0.004966492 0.004131426
#> [15,20)  0.007483755 0.004482710 0.003230170 0.003111629 0.003174983
#> [20,25)  0.020077568 0.013580257 0.008110496 0.005843680 0.006025060
#> [25,30)  0.016935589 0.016679920 0.011217641 0.006306811 0.005210754
#> [30,35)  0.012774474 0.015590870 0.014939049 0.009275413 0.006045883
#> [35,40)  0.008984976 0.009751020 0.011686091 0.010265567 0.007194698
#> [40,45)  0.008236295 0.006798815 0.007273824 0.007939333 0.007681265
#> [45,50)  0.011039686 0.007285606 0.005858061 0.005608974 0.006574370
#> [50,55)  0.013025982 0.009950784 0.006375900 0.004512093 0.004561259
#> [55,60)  0.007437612 0.013162617 0.009941931 0.005595205 0.004082303
#> [60,65)  0.003531416 0.007212728 0.013457331 0.009140948 0.005246936
#> [65,70)  0.001794652 0.002723602 0.006090083 0.010589107 0.007391158
#> [70,75)  0.001398538 0.001344935 0.002249322 0.004709785 0.008167041
#> [75,80)  0.001424558 0.001029161 0.001113015 0.001739506 0.003674485
#> [80,Inf) 0.002163584 0.002092911 0.001832943 0.001695636 0.002295993
#>              [75,80)    [80,Inf)
#> [0,5)    0.006428092 0.007818568
#> [5,10)   0.004394652 0.006668624
#> [10,15)  0.003499930 0.005063919
#> [15,20)  0.002842044 0.003375536
#> [20,25)  0.006119034 0.006834205
#> [25,30)  0.005312829 0.006411113
#> [30,35)  0.005170556 0.006358634
#> [35,40)  0.005057318 0.005075684
#> [40,45)  0.005954503 0.004634304
#> [45,50)  0.006937284 0.005984052
#> [50,55)  0.005634669 0.007313811
#> [55,60)  0.004229362 0.007572295
#> [60,65)  0.003781910 0.006034315
#> [65,70)  0.004086089 0.003724481
#> [70,75)  0.005538787 0.002919180
#> [75,80)  0.006189356 0.003470817
#> [80,Inf) 0.004368850 0.007754153

```

File: R/apply_vaccination.R [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#' Apply vaccination effects to next generation contact matrices
+#' @param ngm next generation matrices. See [generate_ngm()] for creating next generation matrices of a state or a local government area for specific age groups
+#' @param data data set with location specific information on vaccine coverage, efficacy of acquisition/susceptibility and efficacy of transmission/infectiousness for the ordered age groups from lowest to highest of the next generation matrix
```
Comment (njtierney) [inline R/apply_vaccination.R:3]: change "data set" to "data frame"

File: R/apply_vaccination.R [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#' Apply vaccination effects to next generation contact matrices
+#' @param ngm next generation matrices. See [generate_ngm()] for creating next generation matrices of a state or a local government area for specific age groups
+#' @param data data set with location specific information on vaccine coverage, efficacy of acquisition/susceptibility and efficacy of transmission/infectiousness for the ordered age groups from lowest to highest of the next generation matrix 
+#' @param coverage_col bare variable name for the column with information on vaccine coverage by age groups
+#' @param acquisition_col bare variable name for the column with information on efficacy of acquisition
+#' @param transmission_col bare variable name for the column with information on efficacy of transmission
+#' @export
+#' @examples
+#' 
+#' vaccination_effect_example_data
```
Comment (njtierney) [inline R/apply_vaccination.R:10]: Write a comment describing what "vaccination_effect_example_data" is

**Review by njtierney — CHANGES_REQUESTED** [review]
Add a test for `apply_vaccination` to ensure it returns the right output and class

File: R/apply_vaccination.R [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            acquisition_multiplier,
+            transmission_multiplier,
+            FUN = "*"
+          )
+        ),
+      .groups = "drop"
+    ) %>%
+   dplyr::pull(transmission_reduction_matrix)
+  
+ngm_vaccinated=Map('*',ngm,transmission_reduction_matrix)
```
Comment (njtierney) [inline R/apply_vaccination.R:66]: Can you use `<-` instead of `=`, and also add a space around the `<-`

File: R/apply_vaccination.R [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+#' @return list of contact matrices with reduction in transmission matching the next generation matrices
+#' @export
+#' @examples
+#' 
+#' # example data frame with vaccine coverage, acquisition and transmission efficacy 
+#' # of different age groups
+#' vaccination_effect_example_data
+#' 
+#' # Generate next generation matrices
+#' ngm_NSW <- generate_ngm(state_name = "NSW",
```
Comment (njtierney) [inline R/apply_vaccination.R:18]: change to `ngm_nsw` instead of `NSW`

File: R/apply_vaccination.R [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#' vaccination_effect_example_data
+#' 
+#' # Generate next generation matrices
+#' ngm_NSW <- generate_ngm(state_name = "NSW", 
+#'                         age_breaks = c(seq(0, 80, by = 5),Inf),
+#'                         R_target = 1.5
+#'                         ) 
+#'                         
+#'# Apply vaccination effect to next generation matrices                       
+#' ngm_NSW_vacc <- apply_vaccination(ngm = ngm_NSW,
```
Comment (njtierney) [inline R/apply_vaccination.R:24]: change `ngm_NSW` to `ngm_nsw`

**Review by njtierney — CHANGES_REQUESTED** [review]
Just a couple of super minor syntax changes, otherwise looks great!

**Review by njtierney — APPROVED** [review]

**Comment by njtierney** [comment]
Looks good to me!

**Pull request merged**